src/components/interaction-designer/block/BlockToolbar.vue

```diff
@@ -0,0 +1,177 @@
+        v-if="isBlockSelected"
+        v-b-tooltip.hover="trans('flow-builder.deselect-block')"
+        :icon="['far', 'check-circle']"
+        class="fa-btn text-info"
+        @click="isEditable && block_deselect({ blockId: block.uuid })" />
+      <font-awesome-icon
+        v-if="!isBlockSelected"
+        v-b-tooltip.hover="trans('flow-builder.select-block')"
+        :icon="['far', 'circle']"
+        class="fa-btn"
+        @click="isEditable && block_select({ blockId: block.uuid })" />
+    </div>
+    <div class="header-actions-right d-flex">
+      <!--Delete-->
+      <div
+        v-if="isEditable"
+        class="mr-1 ml-2">
+        <div v-if="isDeleting">
+          <button
+            class="btn btn-light btn-xs"
+            @click.prevent="isDeleting = false">
+            <small>{{ trans('flow-builder.cancel') }}</small>
+          </button>
+          <button
+            class="btn btn-danger btn-xs ml-1"
+            @click.prevent="handleDeleteBlock">
+            <small>{{ trans('flow-builder.delete-block') }}</small>
+          </button>
+        </div>
+        <font-awesome-icon
+          v-if="!isDeleting"
+          v-b-tooltip.hover="trans('flow-builder.tooltip-delete-block')"
+          :icon="['far', 'trash-alt']"
+          class="fa-btn text-danger"
+          @click.prevent="isDeleting = true" />
+      </div>
+      <!--Duplicate-->
+      <div class="mr-1 ml-2">
+        <font-awesome-icon
+          v-if="isEditable"
+          v-b-tooltip.hover="trans('flow-builder.tooltip-duplicate-block')"
+          :icon="['fac', 'copy']"
+          class="fa-btn"
+          @click.prevent="handleDuplicateBlock" />
+      </div>
+      <!--Expand block editor-->
+      <div class="mr-1 ml-2">
+        <font-awesome-icon
+          v-if="isEditable"
+          v-b-tooltip.hover="trans('flow-builder.toggle-block-editor-tooltip')"
+          :icon="isEditorVisible ? ['fac', 'minimize'] : ['fac', 'expand']"
+          class="fa-btn"
+          @click.prevent="handleExpandMinimizeBlockEditor" />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+
+import Component, {mixins} from 'vue-class-component'
+import Lang from '@/lib/filters/lang'
+import {Prop} from 'vue-property-decorator'
+import {IBlock} from '@floip/flow-runner'
+import {namespace} from 'vuex-class'
+
+const builderVuexNamespace = namespace('builder')
+const flowVuexNamespace = namespace('flow')
+const validationVuexNamespace = namespace('validation')
+
+@Component({})
+class BlockToolbar extends mixins(Lang) {
+  @Prop() readonly block!: IBlock
+  @Prop() readonly isBlockSelected!: boolean
+  @Prop() readonly isEditorVisible!: boolean
+
+  isDeleting = false
+
+  handleDeleteBlock() {
+    this.block_deselect({blockId: this.block.uuid})
+    this.flow_removeBlock({blockId: this.block.uuid})
+    this.removeValidationStatusesFor({key: `block/${this.block.uuid}`})
+    this.isDeleting = false
+    this.$emit('delete')
```

Comment:
Can we rename this event to be a bit more specific? Especially for destructive actions, it can be important to say `afterDelete()` or if the action isn't actually performed in this component and something should be acted upon from the parent component then `handleDelete()`. In this case, are we consuming this event from anywhere? Should we remove if it's not used?

Reply:
Thanks for that, I'm making a change